### PR TITLE
UI for Search page

### LINKED ISSUE #55: UI for search page

### PR DESCRIPTION
**Pull request by romaGri**
#55 closed

### CODE REVIEW

File: ecommers/i18n/en-US.json [hunk 1]
```diff
@@ -52,15 +54,19 @@
...
     "category": "CATEGORY",
     "fitting": "FITTING",
     "ok": "ok",
+    "recommended":"RECOMMENDED",
+    "refresh":"REFRESH",
     "emailError": "email is not valid.",
     "passwordError": "password should have at least 4 characters.",
     "fieldError": "field should not be empty.",
     "empty": ""
+
```
Comment (amarchenko25) [inline ecommers/i18n/en-US.json]: remove blank line

File: ecommers/lib/ui/pages/search_page.dart [hunk 2]
```diff
@@ -1,8 +1,87 @@
...
 import 'package:flutter/material.dart';
 
 class SearchPage extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
-    return const Text('Search');
+    final localization = I18n.of(context);
+    return SingleChildScrollView(
+      child: Padding(
+        padding: const EdgeInsets.only(left: 20.0, right: 20.0),
```
Comment (amarchenko25) [inline ecommers/lib/ui/pages/search_page.dart]: use the symmetric constructor instead
use Insets class

File: ecommers/lib/ui/pages/search_page.dart [hunk 3]
```diff
@@ -1,8 +1,87 @@
...
+          children: <Widget>[
+            Row(
+              children: <Widget>[
+                Text(
+                  localization.searchTitle,
+                  style: Theme.of(context).textTheme.headline6,
+                ),
+              ],
+            ),
+            const SizedBox(height: 20),
```
Comment (amarchenko25) [inline ecommers/lib/ui/pages/search_page.dart]: Use Insets

File: ecommers/lib/ui/pages/search_page.dart [hunk 4]
```diff
@@ -1,8 +1,87 @@
...
+            SearchTextFiled(),
+            const SizedBox(height: 20),
```
Comment (amarchenko25) [inline ecommers/lib/ui/pages/search_page.dart]: Use Insets

File: ecommers/lib/ui/pages/search_page.dart [hunk 5]
```diff
@@ -1,8 +1,87 @@
...
+                  child: Text(
+                    localization.clear,
+                    style: Theme.of(context).textTheme.headline5.copyWith(
+                        fontSize: FontSizes.small_3x,
+                        color: BrandingColors.primary),
+                  ),
+                ),
+              ],
+            ),
+            const SizedBox(height: 20),
```
Comment (amarchenko25) [inline ecommers/lib/ui/pages/search_page.dart]: Use Insets

Comment (amarchenko25) [reply ecommers/lib/ui/pages/search_page.dart]: check all hardcoded numbers

File: ecommers/lib/ui/widgets/recently_viewed_widget/recently_viewed_item.dart [hunk 6]
```diff
@@ -0,0 +1,58 @@
+import 'package:ecommers/ui/decorations/index.dart';
+import 'package:ecommers/ui/widgets/recently_viewed_widget/index.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+
+class RecentlyViewedItem extends StatelessWidget {
```
Comment (amarchenko25) [inline ecommers/lib/ui/widgets/recently_viewed_widget/recently_viewed_item.dart]: consider using ProductItemSmall instead

File: ecommers/lib/ui/widgets/recently_viewed_widget/recently_viewed_list.dart [hunk 7]
```diff
@@ -0,0 +1,24 @@
+import 'package:ecommers/ui/decorations/dimens/index.dart';
+import 'package:ecommers/ui/widgets/recently_viewed_widget/index.dart';
+import 'package:flutter/cupertino.dart';
+
+class RcentlyViewedList extends StatelessWidget {
```
Comment (amarchenko25) [inline ecommers/lib/ui/widgets/recently_viewed_widget/recently_viewed_list.dart]: please consider moving all content inside RecentlyViewedWidget
And RcentlyViewedList going to be unnecessary

File: ecommers/lib/ui/widgets/search/search_text_filed.dart [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+                border: InputBorder.none,
+                hintText: I18n.of(context).searchHintText,
+                hintStyle: Theme.of(context).textTheme.bodyText1.copyWith(
+                      fontWeight: FontWeight.w300,
+                    ),
+                prefixIcon: Padding(
+                  padding: const EdgeInsetsDirectional.only(start: 100, end: 5),
+                  child: Icon(
+                    Icons.search,
+                    size: 15,
```
Comment (amarchenko25) [inline ecommers/lib/ui/widgets/search/search_text_filed.dart]: use FontSizes class

File: ecommers/lib/ui/widgets/search_recommended_widget/search_recommended_widget.dart [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+    );
+  }
+
+  Widget createItem(String searchRecommendedItem, BuildContext context) {
+    return Container(
+      height: 30,
+      decoration: BoxDecoration(
+        color: BrandingColors.background,
+        borderRadius: const BorderRadius.all(
+          Radius.circular(5.0),
```
Comment (amarchenko25) [inline ecommers/lib/ui/widgets/search_recommended_widget/search_recommended_widget.dart]: Use Radiuses

File: ecommers/lib/ui/pages/search_page.dart [hunk 10]
```diff
@@ -1,8 +1,94 @@
...
+import 'package:ecommers/ui/widgets/search/index.dart';
+import 'package:ecommers/ui/widgets/search_recommended_widget/index.dart';
 import 'package:flutter/material.dart';
 
 class SearchPage extends StatelessWidget {
+final  double recentlyViewedHeight = 55.0;
+final  double recommendedWiHeight = 55.0;
+
+
+
```
Comment (amarchenko25) [inline ecommers/lib/ui/pages/search_page.dart:12]: please remove blank lines

File: ecommers/lib/ui/pages/search_page.dart [hunk 11]
```diff
@@ -1,8 +1,94 @@
...
   @override
+
+
   Widget build(BuildContext context) {
-    return const Text('Search');
+    final localization = I18n.of(context);
+    return SingleChildScrollView(
+      child: Padding(
+        padding: const EdgeInsets.symmetric(horizontal:20.0),
```
Comment (amarchenko25) [inline ecommers/lib/ui/pages/search_page.dart]: Please, Use Insets

File: ecommers/lib/ui/widgets/recently_viewed_widget/product_item_small.dart [hunk 12]
```diff
@@ -0,0 +1,60 @@
+import 'package:ecommers/ui/decorations/dimens/index.dart';
+import 'package:ecommers/ui/decorations/index.dart';
+import 'package:ecommers/ui/widgets/recently_viewed_widget/index.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+
+class ProductItemSmall extends StatelessWidget {
```
Comment (amarchenko25) [inline ecommers/lib/ui/widgets/recently_viewed_widget/product_item_small.dart]: U didn't understand me. I meant that such widget already exists and you don't need to create a new one

Comment (amarchenko25) [reply ecommers/lib/ui/widgets/recently_viewed_widget/product_item_small.dart]: lib/ui/widgets/product_item/product_item_small.dart

File: ecommers/lib/ui/widgets/search/search_text_filed.dart [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+  Widget build(BuildContext context) {
+    return Row(
+      children: <Widget>[
+        Expanded(
+          child: Container(
+            height: 35,
+            decoration: BoxDecoration(
+              color: BrandingColors.blur,
+              borderRadius: const BorderRadius.all(
+                Radius.circular(20),
```
Comment (amarchenko25) [inline ecommers/lib/ui/widgets/search/search_text_filed.dart]: The correct radius for searchField is 18.
Add please property with 18 value to the Radiuses class and use it here

File: ecommers/lib/ui/widgets/recently_viewed_widget/product_item_small.dart (hunk 12 above)
Comment (romaGri) [reply ecommers/lib/ui/widgets/recently_viewed_widget/product_item_small.dart]: sorry)

File: ecommers/lib/ui/pages/search_page.dart [hunk 14]
```diff
@@ -1,8 +1,90 @@
+import 'package:ecommers/generated/i18n.dart';
+import 'package:ecommers/ui/decorations/dimens/index.dart';
+import 'package:ecommers/ui/decorations/index.dart';
+import 'package:ecommers/ui/widgets/recently_viewed_widget/index.dart';
+import 'package:ecommers/ui/widgets/search/index.dart';
+import 'package:ecommers/ui/widgets/search_recommended_widget/index.dart';
 import 'package:flutter/material.dart';
 
 class SearchPage extends StatelessWidget {
+  final double recentlyViewedHeight = 56.0;
```
Comment (Platton96) [inline ecommers/lib/ui/pages/search_page.dart]: may be change final to const

File: ecommers/lib/ui/pages/search_page.dart [hunk 15]
```diff
@@ -1,8 +1,90 @@
...
+                    style: Theme.of(context).textTheme.headline5.copyWith(
+                        fontSize: FontSizes.small_3x,
+                        color: BrandingColors.primary),
+                  ),
+                ),
+              ],
+            ),
+            const SizedBox(height: Insets.x5),
+            Container(
+              height: 70,
```
Comment (Platton96) [inline ecommers/lib/ui/pages/search_page.dart]: use 70.0 instead of 70

File: ecommers/lib/ui/widgets/search/search_text_filed.dart [hunk 16]
```diff
@@ -0,0 +1,46 @@
...
+import 'package:flutter/widgets.dart';
+
+class SearchTextFiled extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return Row(
+      children: <Widget>[
+        Expanded(
+          child: Container(
+            height: 35,
```
Comment (Platton96) [inline ecommers/lib/ui/widgets/search/search_text_filed.dart]: uae 35.0

File: ecommers/lib/ui/widgets/search/search_text_filed.dart [hunk 17]
```diff
@@ -0,0 +1,46 @@
...
+            child: TextFormField(
+              textAlign: TextAlign.left,
+              decoration: InputDecoration(
+                border: InputBorder.none,
+                hintText: I18n.of(context).searchHintText,
+                hintStyle: Theme.of(context).textTheme.bodyText2.copyWith(
+                      fontWeight: FontWeight.w300,
+                    ),
+                prefixIcon: Padding(
+                  padding: const EdgeInsetsDirectional.only(start: 100, end: 5),
```
Comment (Platton96) [inline ecommers/lib/ui/widgets/search/search_text_filed.dart]: use Insets, you can add x25

File: ecommers/lib/ui/widgets/search_recommended_widget/search_recommended_widget.dart [hunk 18]
```diff
@@ -0,0 +1,51 @@
...
+            .map((item) => createItem(item.titile, context))
+            .toList(),
+      ),
+    );
+  }
+
+  Widget createItem(String searchRecommendedItem, BuildContext context) {
+    return InkWell(  //TODO: paste value to search filed
+          onTap: () => {},
+          child: Container(
```
Comment (Platton96) [inline ecommers/lib/ui/widgets/search_recommended_widget/search_recommended_widget.dart]: uncorrect text aligment

**Review by amarchenko25 — APPROVED** [review]

**Pull request merged**